### PR TITLE
feat(playground): add default query fn example

### PR DESCRIPTION
**Pull request by hrvbernardic**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngneat/query/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DefaultQueryFunctionPageComponent {
+  useQuery = inject(QueryProvider);
+  selectedPostId$ = new BehaviorSubject<number | null>(null);
+
+  posts$ = this.useQuery<Post[]>(['/posts']).result$;
+  selectedPost$ = this.selectedPostId$.asObservable().pipe(
+    filter(Boolean),
```
Comment (NetanelBasal) [inline packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts:66]: You can use a Subject and remove the filter

File: packages/playground/src/main.ts [hunk 2]
```diff
@@ -35,6 +38,25 @@ bootstrapApplication(AppComponent, {
             });
           },
         },
+    {
+      provide: QUERY_CLIENT_CONFIG,
+      deps: [HttpClient],
```
Comment (NetanelBasal) [inline packages/playground/src/main.ts:43]: You can remove the deps and use inject inside useFactory

File: packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts (hunk 1 above)
Comment (hrvbernardic) [reply packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts:66]: I can use the Subject but without filter it triggers useQuery with null.

Comment (NetanelBasal) [reply packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts:66]: Why null?

Comment (hrvbernardic) [reply packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts:66]: Tried to replicate the example from Tanstack, when selectedId is null we show a list of posts.

Comment (NetanelBasal) [reply packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts:66]: Ok cool

Comment (hrvbernardic) [reply packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts:66]: had to use ReplaySubject

**Comment by NetanelBasal** [comment]
Can you please pull from master?

**Comment by NetanelBasal** [comment]
@hrvbernardic please fix the conflicts and pull master.

**Comment by NetanelBasal** [comment]
@hrvbernardic, it only runs when you don't pass a queryFn right? the rest of the pages work fine.

**Comment by hrvbernardic** [comment]
yeah I tried the others

**Comment by NetanelBasal** [comment]
Merged. What's next :)

**Comment by hrvbernardic** [comment]
haha, Ill take a look

**Pull request merged**